### PR TITLE
Solana impl in backend

### LINKED ISSUE #83: Solana implementation Backend
## User story
1. As a Poweruser
2. I want / need to mint on the Solana Blockchain
3. So that i can showcase the minting process with Solana

## Acceptance criteria
* Implement Solana in the backend
* Do reseach on which parameters are needed to to connect it to the frontend

## Definition of done (DoD)
* A Feature is Done when all its Acceptance criterias are fulfilled
* A Feauture is Done when the code has been peer reviewed from the dedicated team
* A Feature is Done when it has been tested
* A Feature is Done when its core functionality has been commented properly
* A Feature is Done when it can be fully integrated into the main branch without negative side effect

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by schlingling**
We (@phildree, @MPSebastian, @schlingling) extended the backend with an integration of Solana:
- Impl of Solana specific configs
- mintNFT() works now with solana dev net
- NFT-Metadata-JSON gets stored to aerwaves
- as a NFT-Framework we are using the Metaplex NFT System, find more information here: [https://docs.metaplex.com/architecture/contracts](https://docs.metaplex.com/architecture/contracts)
- find more information on NFT Model on Solana here: [https://lorisleiva.com/owning-digital-assets-in-solana/how-nfts-are-represented-in-solana](https://lorisleiva.com/owning-digital-assets-in-solana/how-nfts-are-represented-in-solana)
- IMPORTANT: NFT Concepts on Solana differs slightly from Ethereum NFT concepts

SOLANA NFT MODEL OVERVIEW:
solana-nft-architecture

closes #83

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "editor.formatOnSave": true
+}
```
Comment (tawfeeka) [inline .vscode/settings.json:3]: I think the .vscode folder doesn't need to be versioned either

**Review by tawfeeka — APPROVED** [review]
nice work

File: .vscode/settings.json (hunk 1 above)
Comment (schlingling) [reply .vscode/settings.json]: deleted the file @tawfeeka

File: packages/middleware/src/lib/middleware.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Middleware
```
Comment (lukas-rotsching) [inline packages/middleware/src/lib/middleware.md:1]: what is that file? Doesn't this belong into the design wiki?

**Review by lukas-rotsching — COMMENTED** [review]
👌

File: packages/middleware/src/lib/middleware.md (hunk 2 above)
Comment (schlingling) [reply packages/middleware/src/lib/middleware.md:1]: not sure, how this file ended up here?! @lukas-rotsching 

@MPSebastian Do you know more?

**Pull request merged**